python/ray/dataframe/dataframe.py

```diff
@@ -167,6 +171,27 @@ def __init__(self, data=None, index=None, columns=None, dtype=None,
         if self._dtypes_cache is None:
             self._correct_dtypes()
 
+    def _get_data(self):
+        data = {}
+        data['blocks'] = self._block_partitions
+        data['col_metadata'] = self._col_metadata
+        data['row_metadata'] = self._row_metadata
+        data['columns'] = self.columns
+        data['index'] = self.index
+        data['dtypes'] = self._dtypes_cache
+
+        return data
+
+    def _set_data(self, data):
+        self._block_partitions = data['blocks']
+        self._col_metadata = data['col_metadata']
```

Comment:
Is the dataframe-passing semantic supposed to imply a copy? If so the metadata objects need to be copied.

Reply:
In pandas, this is not handled through a copy. They retain identical references to columns and rows when passing a `DataFrame` back through the constructor. So, I think that this should be fine.